src/types.jl

```diff
@@ -253,22 +252,53 @@ RObject{RealSxp}
 [1] 1 2 3
 ```
 
+\"\"\"
+
 """
-type RObject{S<:Sxp}
-    p::Ptr{S}
-    # used for pre-defined constants
-    function RObject{S}() where S
-        new(C_NULL)
+
+# Compat.jl not yet provide support for the new syntax for inner consturctor
+# https://github.com/JuliaLang/Compat.jl/issues/332
+# and the new syntax cannot be parse in julie v0.5, therefore using string.
+if VERSION < v"0.6.0-"
```

Comment:
use a more specific cutoff for this. run contrib/commit-name.sh at the merge commit for the type system rewrite pr